pypeit/images/mosaic.py

```diff
@@ -0,0 +1,196 @@
+"""
+:class:`~pypeit.datamodel.DataContainer` object to hold mosaic properties.
+
+.. include common links, assuming primary doc root is up one directory
+.. include:: ../include/links.rst
+"""
+import inspect
+from IPython import embed
+
+import numpy as np
+
+from astropy import table
+from astropy.io import fits
+
+from pypeit import datamodel
+from pypeit import io
+from pypeit.images.detector_container import DetectorContainer
+from pypeit import msgs
+
+
+class Mosaic(datamodel.DataContainer):
+    """
+    Class to hold mosaic parameters and the details of the detectors used to
+    construct the mosaic.
+    """
+
+    # Set the version of this class
+    version = '1.0.0'
+
+    # WARNING: None of the keywords in this datamodel should be the same as in
+    # pypeit.images.detector_container.DetectorContainer.
+    datamodel = {'id': dict(otype=int, descr='Mosaic ID number'),
+                 'detectors': dict(otype=np.ndarray, atype=DetectorContainer,
```

Comment:
I prefer the `@property` approach because it forces the `det` tuple to always be consistent with the list of `detectors`.  I waffled on this, though, because it means that `det` has to be constructed every time it's accessed...